packages-internal/test-utils/src/createRenderer.tsx

```diff
@@ -301,7 +301,7 @@ function render(
   );
   const result: MuiRenderResult = {
     ...testingLibraryRenderResult,
-    user: userEvent.setup(),
+    user: userEvent.setup({ delay: null }),
```

Comment:
Going off topic, but would this works instead?

```suggestion
export const userEventInstance = userEvent.setup({ delay: null }),
```

Reply:
I think you want to initialize it per test. At least that's seems to be [recommended](https://testing-library.com/docs/user-event/intro#writing-tests-with-userevent) on their docs. I understand the instance hold state and you can avoid interference this way.

> We discourage rendering or using any userEvent functions outside of the test itself - e.g. in a before/after hook

Reply:
This is what I'm not clear about. In the source, I didn't saw things that suggest is can't be initialized once for all the test, saving time for each test and making the DX simpler as well.

Reply:
Not sure I follow @oliviertassinari. I think we should just follow the Testing Lib recommendation here i.e. setup userEvent once per test.

Reply:
I'm not following, in the docs I see:

> We recommend invoking userEvent.setup() before the component is rendered.

Which I read as: if we don't call it, at all, it's ok but it's better to call it at least once.

Reply:
With Vitest, it could become an issue, depending on how parallelism is implemented. I imagine that it's OK, but it might not.